tb_plugin/torch_tb_profiler/profiler/data.py

```diff
@@ -118,12 +118,33 @@ def _preprocess_file(caches, trace_path):
                     # only replace the N/A without surrounding double quote
                     fout.write(re.sub(r'(?<!")N/A(?!")', "\"N/A\"", str_data))
                     trace_json = json.loads(fout.getvalue())
-
+                    logger.warning("Get JSONDecodeError: %s, Re-encode it to temp file" % e.msg)
+                    json_reencode = True
+
+        # work-around to remove the "Record Window End" events to avoid the huge end timestamp
+        event_list = trace_json["traceEvents"]
+        end_index = None
+        start_index = None
+        for i in reversed(range(len(event_list))):
+            if event_list[i]["name"] == "Record Window End":
+                end_index = i
+            elif event_list[i]["name"].startswith("Iteration Start:"):
```

Comment:
Are you sure Iteration Start is also a problem? Looking at the code, the timestamp for this is actually copied from profiler_kineto.

Reply:
I just want to find the start and end timestamp to remove the end timestamp in case of the duration is larger than one day. I don't remove the Iteration Start.

Reply:
Ah yes - thanks for clarifying.